sql/core/src/test/scala/org/apache/spark/sql/sources/TableScanSuite.scala

```diff
@@ -83,7 +81,7 @@ case class AllDataTypesScan(
         i.toDouble,
         new java.math.BigDecimal(i),
         new java.math.BigDecimal(i),
-        new Date(1970, 1, 1),
+        Date.valueOf("1970-01-01"),
```

Comment:
For those who haven't encountered this bizarre part of the JDK, both `java.util.Date` and `java.sql.Date` have a constructor that takes year as (year - 1970), 0-indexed month and 1-indexed day. It's deprecated because it's nuts. This expression before was actually referring to Feb 1, 3870. `java.sql.Date` provides a fairly friendly `valueOf` factory method instead for this.